src/scalarstats.jl

```diff
@@ -234,15 +234,23 @@ sem{T<:Real}(a::AbstractArray{T}) = sqrt(var(a) / length(a))
 
 # Median absolute deviation
 """
-    mad(v, center=median(x); constant=1.4826)
+    mad(v)
 
-Compute the median absolute deviation of `v`, optionally specifying a
-precomputed median `center`. `constant` provides a scaling factor that
-defaults to 1.4826 for consistent estimation of the standard deviation
-of a normal distribution.
+Compute the median absolute deviation of `v`.
 """
-mad{T<:Real}(v::AbstractArray{T}, args...;arg...) = mad!(copy(v), args...;arg...)
-mad{T<:Real}(v::Range{T}, args...;arg...) = mad!([v;], args...;arg...)
+function mad{T<:Real}(v::AbstractArray{T})
+    m = median(v)
```

Comment:
I see. We should try to come up with a promote rule that doesn't require the computation of the median. Maybe use the [middle](https://github.com/JuliaLang/julia/blob/master/base/statistics.jl#L457-L462) function since that will be the one that will promote the elements in `median`.

I was thinking of `copy_oftype` because if we strat using it systematically outside Base then it's a indication that it ought to be exported.